app/component/itinerary/navigator/NaviContainer.js

```diff
@@ -54,20 +45,22 @@ function NaviContainer(
     }
   }, [mapRef]);
 
-  const { realTimeLegs, time } = useRealtimeLegs(
-    planarLegs,
-    mapRef,
-    relayEnvironment,
-  );
-
-  if (!realTimeLegs.length) {
+  if (!realTimeLegs?.length) {
     return null;
   }
 
+  const arrivalTime = legTime(realTimeLegs[realTimeLegs.length - 1].end);
```

Comment:
We have last leg available already here, no need to calculate it again. Just use `lastLeg.end`